python/plugins/processing/algs/qgis/Climb.py

```diff
@@ -0,0 +1,260 @@
+# -*- coding: utf-8 -*-
+
+"""
+/***************************************************************************
+ Climb
+        begin                : 2019-05-15
+        copyright            : (C) 2019 by Håvard Tveite
+        email                : havard.tveite@nmbu.no
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+"""
+
+__author__ = 'Håvard Tveite'
+__date__ = '2019-03-01'
+__copyright__ = '(C) 2019 by Håvard Tveite'
+
+# This will get replaced with a git SHA1 when you do a git archive
+
+__revision__ = '$Format:%H$'
+
+import os
+import math
+
+from qgis.PyQt.QtGui import QIcon
+from qgis.PyQt.QtCore import QVariant
+
+from qgis.core import (QgsProcessing,
+                       QgsFeatureSink,
+                       QgsProcessingAlgorithm,
+                       QgsProcessingParameterFeatureSource,
+                       QgsProcessingParameterFeatureSink,
+                       QgsProcessingOutputNumber,
+                       QgsProcessingException,
+                       QgsProcessingUtils,
+                       QgsWkbTypes,
+                       QgsFields,
+                       QgsField)
+
+from processing.algs.qgis.QgisAlgorithm import QgisAlgorithm
+
+
+class Climb(QgisAlgorithm):
+
+    INPUT = 'INPUT'
+    OUTPUT = 'OUTPUT'
+
+    TOTALCLIMB = 'TOTALCLIMB'
+    TOTALDESCENT = 'TOTALDESCENT'
+    MINELEVATION = 'MINELEVATION'
+    MAXELEVATION = 'MAXELEVATION'
+    CLIMBATTRIBUTE = 'climb'
+    DESCENTATTRIBUTE = 'descent'
+    MINELEVATTRIBUTE = 'minelev'
+    MAXELEVATTRIBUTE = 'maxelev'
+
+    def name(self):
+        return 'climbalongline'
+
+    def displayName(self):
+        return self.tr('Climb Along Line')
+
+    def group(self):
+        return self.tr('Vector analysis')
+
+    def groupId(self):
+        return 'vectoranalysis'
+
+    def __init__(self):
+        super().__init__()
+
+    def initAlgorithm(self, config=None):
+        self.addParameter(
+            QgsProcessingParameterFeatureSource(
+                self.INPUT,
+                self.tr('Input (line) layer'),
```

Comment:
```suggestion
                self.tr('Line layer'),
```

AFAIK, we don't have parenthesis in processing labels